cmake/MPIDynamicHelper.cmake

```diff
@@ -2,19 +2,68 @@
 # Configure support for dynamic MPI initialization
 # =============================================================================
 # ~~~
-# TODO : verify that NRN_ENABLE_MPI_DYNAMIC is valid and determine an include
+# Verify that NRN_ENABLE_MPI_DYNAMIC is valid and determine an include
 # directory for each MPI package building libnrnmpi_<mpipkg>.so.
 # Depending on the MPIs used NRNMPI_INCLUDE_<mpipkg> will be defined.
-# Currently dynamic MPI support is not implemented.
+# A number of other lists needed to construct libnrnmpi_<mpipkg>.so
+# are also constructe. The lists are all parallel in the sense that
+# corresponding elements are related to the same mpi installation.
 # ~~~
 
+set(NRN_MPI_BIN_LIST "" CACHE INTERNAL "" FORCE)
+set(NRN_MPI_INCLUDE_LIST "" CACHE INTERNAL "" FORCE)
+set(NRN_MPI_LIBNAME_LIST "" CACHE INTERNAL "" FORCE)
+set(NRN_MPI_TYPE_LIST "" CACHE INTERNAL "" FORCE)
+
+
 if(NRN_ENABLE_MPI)
   if(NRN_ENABLE_MPI_DYNAMIC)
-    if("${NRN_MPI_DYNAMIC}" STREQUAL "")
-      set(NRNMPI_DYNAMICLOAD 1)
-    else()
-      message(FATAL_ERROR "NRN_ENABLE_MPI_DYNAMIC doesn't support multiple MPI yet")
+
+    set(NRNMPI_DYNAMICLOAD 1)
+
+    # compute the NRN_MPI_INCLUDE_LIST and NRN_MPI_BIN_LIST
+    if("${NRN_MPI_DYNAMIC}" STREQUAL "") # use the MPI already found
+      list(APPEND NRN_MPI_INCLUDE_LIST "${MPI_C_HEADER_DIR}")
+      string(REGEX REPLACE "/[^/]*$" "/bin" foo "${MPI_C_HEADER_DIR}")
```

Comment:
On the command line if I specify path with trailing backslash i.e. as `/xxx/bin/` then it doesn't accept.

Reply:
I guess we could replace the regex by ```/[^/]*/*$"``` to handle an optional last '/' character. It would also accept ```/xxx/bin//////```